src/stamina/_core.py

```diff
@@ -384,12 +393,20 @@ def from_params(
         args: tuple[object, ...],
         kw: dict[str, object],
     ) -> _RetryContextIterator:
+        if (
+            isinstance(on, type)
+            and issubclass(on, BaseException)
+            or isinstance(on, tuple)
+        ):
+            _retry = _t.retry_if_exception_type(on)
+        else:
+            _retry = _t.retry_if_exception(on)
```

Comment:
Doing 3 type checks seems more complex than the example in the issue?
```suggestion
        if isinstance(on, Callable):
            _retry = _t.retry_if_exception(on)
        else:
            _retry = _t.retry_if_exception(lambda exc: isinstance(exc, on))
```

Reply:
This doesn't work (correctly), classes are callables too.

Reply:
Hmm, true 🤔
I guess that's about the cleanest way then...